src/WebJobs.Extensions.DurableTask/DurableHttpRequest.cs

```diff
@@ -169,8 +169,11 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                 }
                 else
                 {
-                    // Don't know how to serialize this - use default behavior
+                    // Don't know how to serialize this - use default behavior, forcing TypeNameHandling.Objects to correctly serialize ITokenSource
+                    var currTypeNameSetting = serializer.TypeNameHandling;
+                    serializer.TypeNameHandling = TypeNameHandling.Objects;
```

Comment:
I'm worried this could cause race conditions. If a shared serializer is passed into this method then we're changing the serialization behavior potentially for many different parts of the code. It's only temporary because we're changing it back, but it still seems pretty dangerous to do this and could make debugging serialization problems extremely difficult. A safer alternative would be to create a new serializer object (copying any settings that we think are relevant) and *not* modify the serializer passed in as an argument.

Reply:
Since this was merged, I can clean this up in a quick PR.

Reply:
https://github.com/Azure/azure-functions-durable-extension/pull/1283